### PR TITLE
Version 1.3 release candidate

### PR DESCRIPTION
**Pull request by w-k-jones**
Merging changes for the version 1.3 release from dev into main.

_**Changelog:**_

**Enhancements**

- Significant performance improvements for tracking #89 
- Significant performance improvements for feature detection and segmentation #90 
- Performance improvement for `calculate_area` #93 
- Added ability to set a user defined stub cell value instead of `np.nan`. Default value is `-1` and stub cell values are now integers instead of floats by default #74 
- Added deprecation warnings for parameters `min_num` in feature detection and `d_min` in tracking, and added exceptions when multiple, incompatible parameters are given (e.g. `d_max` and `v_max`) #107 

**Bug fixes**

- Fixed level parameter in segmentation, as this previously had no effect #92 
- Remove `is` comparisons for string literals #99 
- Added missing `raise` for exception in `get_spacings` #105 
- Remove automatic setting of matplotlib backend to `agg` on import #100 
- Fix deprecation warnings for changed import paths in dependencies #110 

**Documentation**

- Added recommended python style #72 
- Updated author list and email addresses #109 

**Repository enhancements**

- Black formatting of all python code and formatting check in actions #78 
- Pre-commit hook for black formatting #96 

Please let me know if there are any addition issues/PRs that need adding before updating the main branch!

### CODE REVIEW

**Comment by w-k-jones** [comment]
_**Changelog:**_

**Enhancements**

- Significant performance improvements for tracking #89 
- Significant performance improvements for feature detection and segmentation #90 
- Performance improvement for `calculate_area` #93 
- Added ability to set a user defined stub cell value instead of `np.nan`. Default value is `-1` and stub cell values are now integers instead of floats by default #74 
- Added deprecation warnings for parameters `min_num` in feature detection and `d_min` in tracking, and added exceptions when multiple, incompatible parameters are given (e.g. `d_max` and `v_max`) #107 

**Bug fixes**

- Fixed level parameter in segmentation, as this previously had no effect #92 
- Remove `is` comparisons for string literals #99 
- Added missing `raise` for exception in `get_spacings` #105 
- Remove automatic setting of matplotlib backend to `agg` on import #100 
- Fix deprecation warnings for changed import paths in dependencies #110 

**Documentation**

- Added recommended python style #72 
- Updated author list and email addresses #109 

**Repository enhancements**

- Black formatting of all python code and formatting check in actions #78 
- Pre-commit hook for black formatting #96

File: setup.py [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+    ],
+    author_email=[
+        "max.heikenfeld@physics.ox.ac.uk",
+        "william.jones@physics.ox.ac.uk",
+        "senf@tropos.de",
+        "sean.freeman@colostate.edu",
+        "julia.kukulies@gu.se",
+        "peter.marinescu@colostate.edu",
+    ],
+    license="GNU",
```
Comment (freemansw1) [inline setup.py:24]: I'm not sure if this is worth stopping the release over, but this is not correct in `main` either. Our license per the `LICENSE` file is the BSD 3-clause license.

**Review by freemansw1 — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (w-k-jones) [reply setup.py:24]: Good catch, I'll fix that quickly

**Review by JuliaKukulies — APPROVED** [review]
Looks all good to me - good job! Also tested the pre-commit since I somehow missed that one, but that was a nice addition and works well.

**Comment by JuliaKukulies** [comment]
I just thought of a small thing - will you add the changelog in the repo or do you think it is enough to have it here? Could be nice for users with a small release note and overview what is different from the previous tobac version. But maybe that is not as important for the minor releases as for the bigger ones.

**Comment by w-k-jones** [comment]
Good point, I'll create and add a changelog.md file that we can continue updating in future

**Pull request merged**